BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m

```diff
@@ -65,7 +65,12 @@ - (NSURLRequest *)urlRequest
 - (void)performRequest
 {
     [self.operation.APIRequest setValue:[self userAgent] forHTTPHeaderField:@"User-Agent"];
-    [self.queueManager enqueueOperation:self.operation];
+    if (self.shouldPerformRequestImmediately == YES) {
+        self.operation.shouldStartImmediately = YES;
```

Comment:
Interesting we have to tell the operation it can start immediately right before we tell it to start.  Shouldn't the latter be enough?

Reply:
operation's start does not start the operation immediately, it puts operation execution on a global thread.